src/main/java/ca/pjer/logback/AwsLogsAppender.java

```diff
@@ -112,24 +115,39 @@ public synchronized void stop() {
             layout.stop();
         }
     }
+    
+    private final List<InputLogEvent> backLog = new LinkedList<InputLogEvent>();
 
     @Override
     protected void append(ILoggingEvent event) {
+        final InputLogEvent inputLogEvent = new InputLogEvent()
+                .withTimestamp(event.getTimeStamp())
+                .withMessage(layout.doLayout(event));
         try {
-            PutLogEventsRequest request = new PutLogEventsRequest()
+            
+            final PutLogEventsRequest request = new PutLogEventsRequest()
                     .withLogGroupName(logGroupName)
                     .withLogStreamName(logStreamName)
-                    .withSequenceToken(sequenceToken)
-                    .withLogEvents(new InputLogEvent()
-                            .withTimestamp(event.getTimeStamp())
-                            .withMessage(layout.doLayout(event)));
-            PutLogEventsResult result = awsLogs.putLogEvents(request);
+                    .withSequenceToken(sequenceToken);
+            if (backLog.size() > 0) {
+                request.withLogEvents(backLog);
+            }
+            request.withLogEvents(inputLogEvent);
+            final PutLogEventsResult result = awsLogs.putLogEvents(request);
             sequenceToken = result.getNextSequenceToken();
+            backLog.clear();
         } catch (DataAlreadyAcceptedException e) {
             sequenceToken = e.getExpectedSequenceToken();
         } catch (InvalidSequenceTokenException e) {
             sequenceToken = e.getExpectedSequenceToken();
             append(event);
+        } catch (AmazonServiceException e) {
+            backLog.add(inputLogEvent);
```

Comment:
I guess we could use `com.amazonaws.services.logs.model.LimitExceededException` ?

I actually have never saw a throttling error in my apps, so its only a guess.